### DISCUSSION TITLE
How to push() an object into where.AND - Enumerable<xyzWhereInput>

### DISCUSSION THREAD

**Discussion by stylemate** [prisma/prisma#11841] [Q&A] [answered]
I'm using NestJS and typescript with Prisma Client 3.9.2

I'm trying to make search system so that:
If query string is `?search=user:John,history`,

After splitting with comma as a delimiter, it will consider strings with ":" as a special string, and non-colon string will be a plain search string.

The structure of `Prisma.xyzFindManyArgs` would be:
```
query = {
    where: {
        OR: [
            {
                fullName: {
                    contains: plainSearchString || '',
                    mode: 'insensitive'
                }
            },
            {
                hobby: {
                    contains: plainSearchString || '',
                    mode: 'insensitive'
                }
            }
        ],
        AND: []
    }
};
```
as a base query so that given plain string query `history` will be searched on desired columns I specify.

If there are queries with colon, like `user:john` I want to push an object:
```
fullName: {
    contains: searchString.split(':')[1] || '',
    mode: 'insensitive'
}
```
into `query.where.AND`.

But `AND`, which is an exported type of Enumerable<xyzWhereInput> doesn't have property of 'push'.
It will throw an error:
```
Property 'push' does not exist on type 'Enumerable<xyzWhereInput>'.
Property 'push' does not exist on type 'xyzWhereInput'
```

This applies to all AND, OR, NOT defined in index.d.ts generated with `prisma generate`.
I've tried chaning the type from `Enumberable<xyzWhereInput>` to `Array<xyzWhereInput>` to see if I can use `push()` method on AND, OR, NOT, but no avail.

I've tried using `Object.assign()`, but it would overwrite the first element of an array (or enumerable?)

Is there a way to append my desired where object into those Enumerables?

Thank you very much

**Comment by stylemate** [comment, marked as answer]
UPDATE:
using `// @ts-ignore` on `query.where.AND.push()` works, but it looks hacky.

Is there a sophisticated way for solving this?

**Reply by tarazena** [reply]
why not create a variable outside the query for AND (like `const AND = []` and then push the data tit, then use `AND` inside the query?

**Reply by stylemate** [reply]
lol, that works.

`const whereAnd: Array<Prisma.xyzWhereInput> = [];`

`whereAnd.push({});`

```
query = {
    where: {
        OR: whereOr,
        AND: whereAnd
    },
    ...
    ...
```

Thank you

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
